docs/user_guide/mitigation.rst

```diff
@@ -80,11 +81,408 @@ Reductions
+    >>> import numpy as np
+    >>> dp = DemographicParity(difference_bound=0.01)
+    >>> X                  = np.array([[0], [1], [2], [3], [4], [5], [6], [7], [8], [9]])
+    >>> y_true             = np.array([ 1 ,  1 ,  1 ,  1 ,  0,   0 ,  0 ,  0 ,  0 ,  0 ])
+    >>> y_pred             = np.array([ 1 ,  1 ,  1 ,  1 ,  0,   0 ,  0 ,  0 ,  0 ,  0 ])
+    >>> sensitive_features = np.array(["a", "b", "a", "a", "b", "a", "b", "b", "a", "b"])
+    >>> selection_rate_group_summary(y_true, y_pred, sensitive_features=sensitive_features)
+    {'overall': 0.4, 'by_group': {'a': 0.6, 'b': 0.2}}
+    >>> dp.load_data(X, y_true, sensitive_features=sensitive_features)
+    >>> dp.gamma(lambda X: y_pred)
+    sign  event  group_id
+    +     all    a           0.2
+                 b          -0.2
+    -     all    a          -0.2
+                 b           0.2
+    dtype: float64
+ 
+Rewriting this into a ratio-based constraint with :code:`ratio_bound`
+:math:`r` we get
+
+.. math::
+
+    r \leq \dfrac{\P[h(X) = y' \given A = a]}{\P[h(X) = y']} \leq \dfrac{1}{r} \; \forall a, y'
+
+Revisiting the same example from above we get
+
+.. doctest:: mitigation
+
+    >>> dp = DemographicParity(ratio_bound=0.9, ratio_bound_slack=0.01)
+    >>> dp.load_data(X, y_pred, sensitive_features=sensitive_features)
+    >>> dp.gamma(lambda X: y_pred)
+    sign  event  group_id
+    +     all    a           0.14
+                 b          -0.22
+    -     all    a          -0.24
+                 b           0.16
+    dtype: float64
+
+Just like :code:`difference_bound` before :code:`ratio_bound_slack` does not
+affect the outcome here, because no mitigation is applied in the example.
+This time the constraint violation is calculated using the :code:`ratio_bound`
+`0.9` and again with the definition from
+:ref:`constraints_binary_classification` and selection rate as the used
+metric:
+
+.. math::
+
+    r \cdot \text{metric}_a - \text{metric}_* = 0.9 \times 0.6 - 0.4 = 0.14 \\
+    r \cdot \text{metric}_b - \text{metric}_* = 0.9 \times 0.2 - 0.4 = -0.22 \\
+    - \text{metric}_a + r \cdot \text{metric}_* = - 0.6 + 0.9 \times 0.4 = -0.24 \\
+    - \text{metric}_b + r \cdot \text{metric}_* = - 0.2 + 0.9 \times 0.4 = 0.16 \\
+
+.. _true_positive_rate_parity:
+.. _false_positive_rate_parity:
+
+True Positive Rate Parity and False Positive Rate Parity
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+A classifier :math:`h(X)` satisfies *True Positive Rate Parity* if
+
+.. math::
+
+    \P[h(X) = 1 \given A = a, Y = 1] = \P[h(X) = 1 \given Y = 1] \; \forall a
+
+and *False Positive Rate Parity* if
+
+.. math::
+
+    \P[h(X) = 1 \given A = a, Y = 0] = \P[h(X) = 1 \given Y = 0] \; \forall a
+
+Note that this formulation of True Positive Rate Parity ignores what happens
+to samples with :math:`Y=0`, and the False Positive Rate Parity ignores
+:math:`Y=1`.
+Refer to :ref:`equalized_odds` for a constraint that extends this by
```

Comment:
Why not introduce equalised odds first, and then point out that TPRP and FPRP are the two halves of it. I feel that referring forward like this is a bit odd.

Reply:
Because then I have to refer the other way forward. I did
```
A
B
C (depends on A, B)
```
and you're suggesting

```
C (depends on A, B which at this point we don't even know)
A
B
```
I really don't see how that's better.

Reply:
No, you'd introduce A, and then note that B and C are subsets of it. In this case, it might just be because I learned about EO first, and so to me it's the foundational one. I suppose it doesn't really matter.

Reply:
Same, and I found it terribly confusing because of that. I would have much preferred hearing about TPRP and FPRP first.